internal/goalstate/goalstatefromvmsettings.go

```diff
@@ -9,21 +9,26 @@ import (
 	"github.com/pkg/errors"
 )
 
-func GetImmediateRunCommandGoalStates(ctx *log.Context, communicator hostgacommunicator.IHostGACommunicator) ([]hostgacommunicator.ExtensionGoalStates, error) {
-	vmSettings, err := communicator.GetImmediateVMSettings(ctx)
+func GetImmediateRunCommandGoalStates(ctx *log.Context, communicator hostgacommunicator.IHostGACommunicator, lastProcessedETag string) ([]hostgacommunicator.ImmediateExtensionGoalState, string, error) {
```

Comment:
nil checks on communicator and lastProcessedEtag ?

Reply:
Added the check on `communicator`. `lastProcessedEtag` does not support passing nil values as string types does not support that.